climate_scanner/sentiment/sentiment_classifier.py

```diff
@@ -0,0 +1,387 @@
+        
+        # Model Predict:
+        self._batch_size_p          = params['predict']['batch_size']
+        self._verbose_p             = params['predict']['verbose']
+        self._steps_p               = params['predict']['steps'] 
+        self._max_queue_size_p      = params['predict']['max_queue_size']
+        self._workers_p             = params['predict']['workers']
+        self._use_multiprocessing_p = params['predict']['use_multiprocessing']
+        
+                 ###############################
+                 ####### MODEL STRUCTURE #######
+                 ###############################
+       
+        
+        # Model structure
+        self.Embedding = tf.keras.layers.Embedding(self._vocab_size,
+                                                   self._embedding_dim,
+                                                   embeddings_initializer=Constant(embedding_matrix),
+                                                   trainable=False)
+        self.LSTM      = tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(self._embedding_dim*2))
+        self.dense1    = tf.keras.layers.Dense(24, activation='relu')
+        self.dense2    = tf.keras.layers.Dense(1, activation='sigmoid')
+        
+    
+    def call(self,inputs):
+        # Method defining the forward pass in the classifier model (called internally during forward pass in compile method)
+        # Args:
+        #   inputs     ->  np.array() representing tokens 
+        # returns:
+        #   lastTensor -> tensor representing last layer in the model    
+        
+        x = self.Embedding(inputs)
+        x = self.LSTM(x)
+        x = self.dense1(x)
+        lastTensor = self.dense2(x)
+        return lastTensor
+    
+    def Compile(self):
+        # Method to compile the model
+        self.compile(optimizer=self._opt, 
+                     loss=self._model_loss, 
+                     metrics=self._model_metrics)
+        
+    def Fit(self,training_data,validation_data=None,call_backs_dir = None):
+        # Method to fit the model to current data
+        # Args:
+        #    training_data   -> tuple(X,y) where X and y are of type numpy.array(). X represents text mapping, y is the label
+        #    validation_data -> tuple(X,y) where X and y are of type numpy.array(). X represents text mapping, y is the label, defaults to None
+        #    call_back_dir   -> optional, of type str describing name of log directory for callbacks, default is None
+        # Returns:
+        #    ---
+        if call_backs_dir:
+            # tensorboard -log files 
+            log_dir = get_data(self._path_to_logs)+call_backs_dir + '\\'+datetime.now().strftime("%Y%m%d-%H%M%S")
+            tensorboard_callback = keras.callbacks.TensorBoard(log_dir=log_dir)
+            
+            # checkpoints to save model
+            checkpoint_dir = get_data(self._path_to_checkpoints) +call_backs_dir +'\\'
```

Comment:
Us os.path.join(), rather than + here, e.g. 

```
log_dir = get_data(os.path.join(self._path_to_logs, call_backs_dir , datetime.now().strftime("%Y%m%d-%H%M%S")))
```